packages/rest/src/parser.ts

```diff
@@ -77,11 +92,30 @@ function buildOperationArguments(
     const rawValue = getParamFromRequest(spec, request, pathParams);
     const coercedValue = coerceParameter(rawValue, spec);
     paramArgs.push(coercedValue);
-  }
 
-  debug('Validating request body - value %j', body);
-  validateRequestBody(body, operationSpec.requestBody, globalSchemas);
+    if (spec.in === 'query' && paramSpec.schema != null) {
+      isQuery = true;
+      paramName = paramSpec.name;
+      paramSchema = paramSpec.schema || [];
+      // tslint:disable-next-line:no-any
+      (<any>queryValue)[paramName] = coercedValue;
+      // tslint:disable-next-line:no-any
+      (<any>schemasValue)[paramName] = paramSchema;
+    }
+  }
 
+  //if query parameters from URL - send to query validation
+  if (isQuery) {
+    query.value = queryValue;
+    globalSchemas = {properties: schemasValue};
+    query.schema = globalSchemas;
+    validateRequestQuery(query, operationSpec.requestBody, globalSchemas);
```

Comment:
IIUC, you are building an object with all parameters from the query string, and then validate this object in one pass.  I have two reservations about this approach:
 - It seems overly complicated to me.
 - There are more parameter sources besides `query` and `body`, we need to validate parameters from all sources (e.g. `path` and `header`).

Have you considered to call `ajv` validator repeatedly, once for each parameter?

Reply:
Hello @bajtos ,
sorry for the delay - yes i do plan to close the issue soon and in regards to your comments:
1. I will do the rebase as requested.
2. Regarding the validation of all parameters all together - i did it in this way due to following same validation standard done for body parameters.
3. Regrading the request for handling other parameter types, i think a new issue should be opened for other parameter types as the initial issue was for query only.